addons/website_slides/models/slide_slide.py

```diff
@@ -569,26 +570,28 @@ def _action_vote(self, upvote=True):
         channel = slide_id.channel_id
         karma_to_add = 0
 
+        reason_lesson_upvoted = _('Lesson upvoted')
+        reason_lesson_downvoted = _('Lesson downvoted')
+
         for slide_partner in slide_partners:
             if upvote:
                 new_vote = 0 if slide_partner.vote == -1 else 1
                 if slide_partner.vote != 1:
-                    karma_to_add += channel.karma_gen_slide_vote
+                    karma_to_add = channel.karma_gen_slide_vote
             else:
                 new_vote = 0 if slide_partner.vote == 1 else -1
                 if slide_partner.vote != -1:
-                    karma_to_add -= channel.karma_gen_slide_vote
+                    karma_to_add = -channel.karma_gen_slide_vote
             slide_partner.vote = new_vote
+            if karma_to_add:
+                self.env.user.add_karma(karma_to_add, slide_partner.channel_id, reason_lesson_upvoted if upvote else reason_lesson_downvoted)
```

Comment:
I don't think karma_to_add is still a useful variable here. 
simply call add karma directly under `if slide_partner.vote != 1 (or -1)`
and while we are at it, do the same to assign slide_partner.vote : 
`slide_partner.vote = 0 if slide_partner.vote == -1 else 1` and same in else. -> of course do this after calling add karma

Reply:
Done!